framework/src/mesh/AnnularMesh.C

```diff
@@ -0,0 +1,230 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "AnnularMesh.h"
+
+#include "libmesh/face_quad4.h"
+#include "libmesh/face_tri3.h"
+
+template <>
+InputParameters
+validParams<AnnularMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRangeCheckedParam<unsigned int>(
+      "nr", 1, "nr>0", "Number of elements in the radial direction");
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "nt", "nt>1", "Number of elements in the angular direction");
+  params.addRequiredRangeCheckedParam<Real>(
+      "rmin",
+      "rmin>=0.0",
+      "Inner radius.  If rmin=0 then a disc mesh (with no central hole) will be created.");
+  params.addRequiredParam<Real>("rmax", "Outer radius");
+  params.addParam<Real>("tmin", 0.0, "Minimum angle, measured anticlockwise from x axis");
+  params.addParam<Real>("tmax",
+                        2 * M_PI,
+                        "Maximum angle, measured anticlockwise from x axis.  If "
+                        "tmin=0 and tmax=2Pi an annular mesh is created.  "
+                        "Otherwise, only a sector of an annulus is created");
```

Comment:
We may need more robust checking on combinations of parameters. It users put in one or two radial elements but use a full radius (instead of a sector) what will the element look like? At the very least we'll have to be careful about inverted elements. Also there needs to be a max range on tmax, right?

Reply:
Constraints on tmax and nt.  Yes, i'll include this.  I think appropriate constraints are

tmax - tmin <= 2 Pi

(then tmin could be negative, or tmax could be > 2Pi, if convenient to users) and

nt > (tmax - tmin) / Pi

Then a full annulus must have >2 angular elements, an almost-full annulus must have >1 angular elements, a half annulus must have >1 elements, a less-than-half annulus must have >0 angular elements.